reservation/actions.go

```diff
@@ -0,0 +1,184 @@
+	reservationRequest, ok := eventCtx.(*InitReservationContext)
+	if !ok {
+		return r.HandleError(fsm.ErrInvalidContextType)
+	}
+
+	keyRes, err := r.cfg.Wallet.DeriveNextKey(r.ctx, KeyFamily)
+	if err != nil {
+		return r.HandleError(err)
+	}
+
+	// Send the client reservation details to the server.
+	log.Debugf("dispatching reservation to server: %x",
+		reservationRequest.ReservationID)
+
+	request := &looprpc.ServerOpenReservationRequest{
+		ReservationId: reservationRequest.ReservationID[:],
+		ClientKey:     keyRes.PubKey.SerializeCompressed(),
+	}
+
+	_, err = r.cfg.ReservationClient.OpenReservation(
+		context.Background(), request,
+	)
+	if err != nil {
+		return r.HandleError(err)
+	}
+
+	reservation := &Reservation{
+		ID:               reservationRequest.ReservationID,
+		Value:            reservationRequest.Value,
+		ClientPubkey:     keyRes.PubKey,
+		ServerPubkey:     reservationRequest.ServerPubkey,
+		KeyLocator:       keyRes.KeyLocator,
+		Expiry:           reservationRequest.Expiry,
+		InitiationHeight: int32(reservationRequest.HeightHint),
+	}
+
+	r.reservation = reservation
+
+	// Create the reservation in the database.
+	err = r.cfg.Store.CreateReservation(r.ctx, reservation)
+	if err != nil {
+		return r.HandleError(err)
+	}
+
+	return OnBroadcast
+}
+
+// SubscribeToConfirmationAction is the action that is executed when the
+// reservation is waiting for confirmation. It subscribes to the confirmation
+// of the reservation transaction.
+func (r *FSM) SubscribeToConfirmationAction(
+	_ fsm.EventContext) fsm.EventType {
+
+	pkscript, err := r.reservation.GetPkScript()
+	if err != nil {
+		return r.HandleError(err)
+	}
+
+	// Subscribe to the confirmation of the reservation transaction.
+	log.Debugf("subscribing to conf for reservation: %x pkscript: %x,"+
+		"initiationheight %v", r.reservation.ID,
+		pkscript, r.reservation.InitiationHeight)
+
+	confChan, errChan, err := r.cfg.ChainNotifier.RegisterConfirmationsNtfn(
+		r.ctx, nil, pkscript,
+		DefaultConfTarget, r.reservation.InitiationHeight,
+	)
+	if err != nil {
+		r.Errorf("unable to subscribe to conf: %v", err)
+		return r.HandleError(err)
+	}
+
+	blockChan, errChan2, err := r.cfg.ChainNotifier.RegisterBlockEpochNtfn(
+		r.ctx,
+	)
+	if err != nil {
+		r.Errorf("unable to subscribe to conf: %v", err)
+		return r.HandleError(err)
+	}
+
+	// We'll now wait for the confirmation of the reservation transaction.
+	for {
+		select {
+		case err := <-errChan:
+			r.Errorf("conf subsciption error: %v", err)
+			return r.HandleError(err)
+
+		case err := <-errChan2:
+			r.Errorf("block subsciption error: %v", err)
+			return r.HandleError(err)
+
+		case confInfo := <-confChan:
+			r.Debugf("reservation confirmed: %v", confInfo)
+			r.reservation.ConfirmationHeight = confInfo.BlockHeight
+
+			outpoint, err := r.reservation.findReservationOutput(
+				confInfo.Tx,
+			)
+			if err != nil {
+				return r.HandleError(err)
+			}
+
+			r.reservation.Outpoint = outpoint
```

Comment:
nit: I guess it's less of an issue here but should we always set members after ensuring that the DB state is updated?